doc/design/zarr-support.md

```diff
@@ -0,0 +1,17 @@
+The Zarr format is being used by NGFF to support the next generation of microscopy data. This involves a potentially 
+nested folder structure with named files inside the folder. In the DANDI this would mean:
+
+1. An asset is associated with a single zarr folder. From a user perspective this is still a single asset and the UI 
+   should not try to delve into the structure of the folder. The CLI should be able to download the entire tree. Matt
+   at kitware is looking into IPFS + NGFF, so we should at least keep that in mind.
+3. Blob store allows for a folder, which contains the zarr named "locations" and data. That is given a root prefix, 
+   a zarr-compliant software can navigate the zarr metadata/structure using relative path rules.
+3. Blob url should continue to point to a single location in an s3 bucket instead of a collection of files.
+4. The key connected with the zarr file should be a tree-hash of dandi-etags of the folder.
```

Comment:
IIRC blob key (well -- `id`) is just a uuid : `blob_id = models.UUIDField(unique=True)` , and we have a dedicated `etag` for content identity. should it be the same here, in that 
```suggestion
4. The `etag` connected with the zarr file should be a tree-hash of dandi-etags of the folder.
```